File: src/org/usfirst/frc/team1678/robot/OI.java
package org.usfirst.frc.team1678.robot;

import org.usfirst.frc.team1678.robot.commands.ShiftCommand;

import edu.wpi.first.wpilibj.Joystick;
import edu.wpi.first.wpilibj.buttons.JoystickButton;

/**
 * This class is the glue that binds the controls on the physical operator
 * interface to the commands and command groups that allow control of the robot.
 */
public class OI {
	private static Joystick driverLeft = new Joystick(RobotMap.leftDriverStick);
	private static Joystick driverRight = new Joystick(RobotMap.rightDriverStick);
	private static JoystickButton shiftLow = new JoystickButton(driverLeft, 3);
	private static JoystickButton shiftHigh = new JoystickButton(driverRight, 3);
	
	static {
		shiftLow.whenPressed(new ShiftCommand(false));
		shiftHigh.whenPressed(new ShiftCommand(true));
	}
	
	public static double getForwardPower() {
		return driverLeft.getRawAxis(1);
	}
	
	public static double getTurning() {
		return driverRight.getRawAxis(0);
	}
}

File: src/OI.cpp
#include "OI.h"
#include <Commands/ShiftCommand.h>
#include <Commands/DriveDistanceCommand.h>

OI::OI()
	: driverRight(new Joystick(RobotMap::rightDriverStick)), driverLeft(new Joystick(RobotMap::leftDriverStick)), manipulator(new Joystick(2)),
	  shiftLow(new JoystickButton(driverLeft.get(), 3)), shiftHigh(new JoystickButton(driverRight.get(), 3)),
	  driveDistance(new JoystickButton(driverRight.get(), 1))
{
	shiftLow->WhenPressed(new ShiftCommand(false));
	shiftHigh->WhenPressed(new ShiftCommand(true));
	driveDistance->WhenPressed(new DriveDistanceCommand(1.5));
}

double OI::getForwardPower()
{
	return -driverLeft->GetRawAxis(1);
}

double OI::getTurningPower()
{
	return -driverRight->GetRawAxis(0);
}

double OI::getElevatorPower()
{
	return -manipulator->GetRawAxis(1);
}
